ClientPlugin/Settings/Elements/Slider.cs

```diff
@@ -0,0 +1,68 @@
+﻿using Sandbox.Graphics.GUI;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace ClientPlugin.Settings.Elements
+{
+    internal class SliderAttribute : Attribute, IElement
+    {
+        public enum SliderType
+        {
+            Integer,
+            Float,
+        }
+
+        public readonly float Min;
+        public readonly float Max;
+        public readonly float Step;
+        public readonly SliderType Type;
+        public readonly string Description;
+
+        public SliderAttribute(float min, float max, float step = 1f, SliderType type = SliderType.Float, string description = null)
+        {
+            Min = min;
+            Max = max;
+            Step = step;
+            Type = type;
+            Description = description;
+        }
+
+        public List<MyGuiControlBase> GetElements(string name, Func<object> propertyGetter, Action<object> propertySetter)
+        {
+            void ValueUpdate(MyGuiControlSlider element)
+            {
+                switch (Type)
+                {
+                    case SliderType.Integer:
+                        propertySetter(Convert.ToInt32(element.Value));
+                        break;
+                    case SliderType.Float:
+                        propertySetter(element.Value);
+                        break;
+                }
+
+            }
+
+            var slider = new MyGuiControlSlider(
```

Comment:
The sliders don't show their current values. It is not shown while being moved either.
![image](https://github.com/user-attachments/assets/fe6c64cd-8101-4b8a-b50a-6fad9e5b6726)

Reply:
Sliders have gotten huge improvements in https://github.com/sepluginloader/PluginTemplate/pull/22/commits/fdbabce7b3c8de2a4b8c0cf3fc5754d291ca3ad2 (this included)